.github/workflows/deploy.yml

```diff
@@ -0,0 +1,81 @@
+name: Deploy
+
+on:
+  push:
+    branches: [ 'develop', 'master' ]
+
+jobs:
+  cancel-previous:
+    name: 'Cancel Previous Runs'
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    steps:
+      - uses: styfle/cancel-workflow-action@0.8.0
+        with:
+          access_token: ${{ github.token }}
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: cancel-previous
+    env:
+      SB_APP_DIR: $HOME/work/switchboad-dapp/switchboad-dapp
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v2-beta
+        with:
+          node-version: '10.22.0'
+          registry-url: https://registry.npmjs.org/
+
+      - name: Setup AWS Host machine
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+
+      - name: Install npm packages
+        run: |
+          npm install
+          envsubst < src/environments/constants.template.ts > src/environments/constants.ts
+
+      - name: Prepare for master branch
+        if: github.ref == 'refs/heads/master'
+        run: |
+          cp src/environments/environment.prod.ts src/environments/environment.ts
+          echo "Prod Env Enabled"
+          npm run build-prod
+
+      - name: Prepare for non-master branch
+        if: github.ref != 'refs/heads/master'
+        run: |
+          cp src/environments/environment.dev.ts src/environments/environment.ts
+          echo "Prod Env Disabled"
+          npm run build
+
+      - name: Run Test
+        run: |
+          npm install
```

Comment:
Is this `npm install` necessary if we already have the "Install npm packages" step

Reply:
It looks like caching is not working. I tried without that. but it was failing. may be I'll check later.